ui/job-view/details/PinBoard.jsx

```diff
@@ -179,22 +180,52 @@ export default class PinBoard extends React.Component {
   }
 
   retriggerAllPinnedJobs() {
-    // pushing pinned jobs to a list.
     const jobIds = Object.keys(this.props.pinnedJobs);
-    const plurality = jobIds.length > 1 ? 's' : '';
-
-    JobModel.retrigger(this.$rootScope.repoName, jobIds)
-      .then((resp) => {
-        if (resp.ok) {
-          this.thNotify.send(
-            `Retrigger request sent for ${jobIds.length} pinned job${plurality}`,
-            'success');
-        } else {
-          throw new Error(formatModelError(resp, `Unable to send retrigger${plurality}`));
+
+    if (!this.props.isLoggedIn) {
+      return this.$timeout(this.thNotify.send('Must be logged in to retrigger jobs', 'danger'));
+    }
+
+    this.$timeout(() => this.thNotify.send(
+      'Request sent to retrigger all pinned jobs via actions.json',
+      'success'),
+    );
+
+    try {
+      jobIds.forEach(async (id) => {
+        const job = await JobModel.get(this.$rootScope.repoName, id);
+        const decisionTaskId = await this.ThResultSetStore.getGeckoDecisionTaskId(job.result_set_id);
+        const results = await TaskclusterModel.load(decisionTaskId, job);
+
+        if (results) {
+          const retriggerTask = results.actions.find(result => result.name === 'retrigger');
+
+          if (retriggerTask) {
+            try {
+              await TaskclusterModel.submit({
+                action: retriggerTask,
+                decisionTaskId,
+                taskId: results.originalTaskId,
+                input: {},
+                staticActionVariables: results.staticActionVariables,
+              });
+            } catch (e) {
+              // The full message is too large to fit in a Treeherder
+              // notification box.
+              this.$timeout(() => this.thNotify.send(
+                formatTaskclusterError(e),
+                'danger',
+                { sticky: true }),
+              );
+            }
+          }
         }
-      })
-      .catch(error => this.$timeout(this.thNotify.send(error, 'danger')))
-      .finally(() => this.$rootScope.$apply());
+      });
+    } catch (e) {
+      this.thNotify.send('Unable to retrigger all jobs', 'danger', { sticky: true });
+    } finally {
+      this.$rootScope.$apply();
```

Comment:
This is pre-existing, I know.  But we should also be able to remove ``this.$rootScope.$apply()`` now as well.  Sorry for the cruft left in here.   These are leftovers from our Angular days that still need to be cleaned up.